web/src/modules/administration/modules/injest-data/components/DataIngestionEditor.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+  <v-dialog
+    v-model="internalVisible"
+    persistent
+    max-width="700"
+  >
+    <v-card>
+      <v-toolbar
+        color="primary"
+        variant="dark"
+        title="Edit Location"
+      >
+        <v-spacer></v-spacer>
+        <v-btn
+          icon
+          @click="handleClose"
+          color="white"
+        >
+          <v-icon>mdi-close</v-icon>
+        </v-btn>
+      </v-toolbar>
+
+      <v-card-text>
+        <v-select
+          v-model="selectedId"
+          :items="dataIngestionSources"
+          item-title="source_name"
+          item-value="id"
+          label="Data Source"
+          dense
+          outlined
+        />
+        <v-file-input
+          v-model="selectedFile"
+          label="Upload File"
+          dense
+          outlined
+          accept=".csv"
+          show-size
+          prepend-icon="mdi-upload"
+          class="mb-2"
+        />
+      </v-card-text>
+
+      <v-card-actions class="mx-4 mb-2">
+        <v-btn
+          color="primary"
+          variant="flat"
+          @click="handleSave"
+          >Save</v-btn
+        >
+        <v-spacer></v-spacer>
+        <v-btn
+          color="yg_sun"
+          variant="outlined"
+          @click="handleClose"
+          >Close</v-btn
+        >
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script setup lang="ts">
+import { storeToRefs } from "pinia";
+import { computed, onMounted } from "vue";
+import { isNil } from "lodash";
+
+import { useUserStore } from "@/store/UserStore";
+import { useDataIngestionSourceAdminStore } from "../store";
+import { useNotificationStore } from "@/store/NotificationStore";
+import { useRouter } from "vue-router";
+
+const props = defineProps<{ modelValue: boolean }>();
+const emit = defineEmits<{ "update:modelValue": [boolean] }>();
+
+const userStore = useUserStore();
+const dataIngestionStore = useDataIngestionSourceAdminStore();
+const notificationStore = useNotificationStore();
+
+const { user } = storeToRefs(userStore);
+const { dataIngestionSources, selectedDataIngestionSourceId, selectedDataIngestionFile } =
+  storeToRefs(dataIngestionStore);
+
+const internalVisible = computed<boolean>({
+  get: () => props.modelValue,
+  set: (modelValue) => emit("update:modelValue", modelValue),
+});
+
+const selectedId = computed<number | undefined>({
+  get: () => selectedDataIngestionSourceId.value,
+  set: (source_id) => {
+    if (source_id != null) dataIngestionStore.selectDataIngestionSourceId(source_id);
```

Comment:
Prefer guard clauses or wrapping "if" content with { } if you have long content.

Also consider writing this effect like:
```vue
<template>
<v-select
  :model-value="selectedDataIngestionSourceId"
  @update:model-value="updateDataIngestionSource"
>
</template>

<script>
function updateDataIngestionsSource(dataIngestionSourceId: number) {
   // some effect
} 
</script>
```
as its more linear and easier to debug later on.